### PR TITLE
Add log for reputation service

### LINKED ISSUE #446: Enhance analyzer results
Goal: The analyser table currently shows what the analyser returned. However it does not show the exact reason why reputation didnt made it to the user.  This is decided in a second step in the reputation service itself. This should be logged as well in the same way.
However it needs to be set into its own table since one message can have multiple attempts for the same message and even the same person.

The results should be available via the analyser embed.

### PR DESCRIPTION
**Pull request by RainbowDashLabs**
**Checklist**
- [x] I made changes to commands
  - [x] I fully localised the command
  - [x] I fully checked the commands functionality

- [x] I added new localisation codes
  - [x] I fully translated it for all languages
  - [x] I sorted properties alphabetically

- [x] I made changes to the database
  - [x] I added my changed to a patch file
  - [x] I made sure my database was up to date
  - [x] I checked that the migration works

- [x] I made changes to internal structure 

**Short Description**
Adds a log for the reputation submission to the analyzer log. Could replace the debug emojis in the long run

Closes #446

### CODE REVIEW

File: src/main/java/de/chojo/repbot/analyzer/results/AnalyzerResult.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import de.chojo.repbot.analyzer.results.match.MatchAnalyzerResult;
+import de.chojo.repbot.analyzer.results.match.ThankType;
+import de.chojo.repbot.analyzer.results.match.fuzzy.MemberMatch;
+import de.chojo.repbot.dao.snapshots.analyzer.ResultSnapshot;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.Message;
+
+import java.util.List;
+
+public interface AnalyzerResult {
```
Comment (Goldmensch) [inline src/main/java/de/chojo/repbot/analyzer/results/AnalyzerResult.java:18]: This interface should be sealed.

File: src/main/java/de/chojo/repbot/dao/access/guild/reputation/sub/Analyzer.java [hunk 2]
```diff
@@ -73,6 +94,25 @@ public Optional<ResultEntry> get(long messageId) {
...
+                    } catch (JsonProcessingException e) {
+                        log.error("Could not deserialize result", e);
+                        throw new SQLException(e);
+                    }
+                    return new SubmitResultEntry(result, row.getLong("channel_id"), messageId, row.getTimestamp("submitted").toInstant());
+                }).allSync();
+
+        if(submitResults.isEmpty() && resultEntry.isEmpty()) return Optional.empty();
+
+
```
Comment (Goldmensch) [inline src/main/java/de/chojo/repbot/dao/access/guild/reputation/sub/Analyzer.java:114]: Delete unecessary line.

**Review by Goldmensch — DISMISSED** [review]

File: src/main/java/de/chojo/repbot/analyzer/results/AnalyzerResult.java (hunk 1 above)
Comment (RainbowDashLabs) [reply src/main/java/de/chojo/repbot/analyzer/results/AnalyzerResult.java:18]: No

File: src/main/resources/locale_pt_BR.properties [hunk 3]
```diff
@@ -380,6 +380,24 @@ selfCleanup.leave=Você não me configurou.\nEu só fico em guildas ativas, que
...
+submitresult.embedsend=Foi enviado um embutido.
+submitresult.nodonorrole=Doador %USER% falta um papel de doador.
+submitresult.noreceiverrole=Usuário %USER% está faltando um papel de receptor.
+submitresult.norecentmembers=Não foram encontrados membros recentes ligados ao autor.
+submitresult.notargets=Nenhum alvo encontrado para mensagem e mensagem de confirmação está desativado.
+submitresult.outdatedmessage=A mensagem está desatualizada.
+submitresult.outdatedreferencemessage=A mensagem de referência está desatualizada.
+submitresult.receiverlimit=Limite do receptor atingido.
+submitresult.selfvote=Tentativa de auto-voto do usuário
+submitresult.submitting=Submeter a reputação do tipo %TYPE% por usuário %USER%% por usuário
```
Comment (Taucher2003) [inline src/main/resources/locale_pt_BR.properties:398]:
```suggestion
submitresult.submitting=Submeter a reputação do tipo %TYPE% por usuário %USER% por usuário
```

**Review by Taucher2003 — APPROVED** [review]

**Review by Goldmensch — APPROVED** [review]

**Pull request merged**